### PR TITLE
Reports list is organzed by pending and submitted reports

### PR DESCRIPTION
**Pull request by cmgiven**
Connects #277 

In addition:
* Updates the nav bar to use a count of unsubmitted reports, per designs
* Refreshes the nav bar after report submit
* Removes unnecessary state from ReportSubmitForm (`useApi` takes care of it)
* Unifies currency formatting and text input validations, closing out some todos left as comments

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/ReportDetail/ReportSubmitForm.tsx [hunk 1]
```diff
@@ -6,123 +6,127 @@ import UserContext from '../../contexts/User/UserContext';
...
-			accredited,
-			c4KRevenue,
-			retroactiveC4KRevenue,
-			familyFeesRevenue,
-			submittedAt
-		};
-	}
+  const params: ApiOrganizationsOrgIdReportsIdPutRequest = {
+    id: report.id || 0,
+    orgId: idx(user, _ => _.orgPermissions[0].organizationId) || 0
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/ReportDetail/ReportSubmitForm.tsx]: use user id util

File: src/Hedwig/ClientApp/src/containers/Reports/Reports.tsx [hunk 2]
```diff
@@ -1,96 +1,114 @@
...
+          <th scope="row">
+            <Link to={`/reports/${row.id}`} className="usa-link">
+              {monthFormatter(row.reportingPeriod.period)}
+            </Link>
+          </th>
+        ),
+        sort: row => row.reportingPeriod.period && row.reportingPeriod.period.getMonth() || -1,
+      },
+      {
+        name: 'Type',
```
Comment (mmazanec22) [inline src/Hedwig/ClientApp/src/containers/Reports/Reports.tsx]: I'm sure there's some ticket about this and that this page isn't done, but just flagging that right now we're showing abbreviations without spelling them out in the content: https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-located.html

Comment (cmgiven) [reply src/Hedwig/ClientApp/src/containers/Reports/Reports.tsx]: I don't think there's a ticket for that. I was thinking about that one. I'll create.

Comment (mmazanec22) [reply src/Hedwig/ClientApp/src/containers/Reports/Reports.tsx]: I'm doing some research rn into legend best practices-- whether we should associate it with the tables in some way, or associate the values with the labels, or just have an aria label for the tags within the table, or what.  Like many data-related questions, it's not abundantly clear in WCAG :/

**Pull request merged**